src/lake/Lake/DSL/Meta.lean

```diff
@@ -53,9 +56,42 @@ scoped syntax (name := metaIf)
 "meta " "if " term " then " cmdDo (" else " cmdDo)? : command
 
 elab_rules : command | `(meta if $c then $t $[else $e?]?) => do
-  if (← withRef c <| runTermElabM fun _ => evalTerm Bool c) then
+  if (← withRef c <| runTermElabM fun _ => evalTerm Bool (toTypeExpr Bool) c .unsafe) then
     let cmd := mkNullNode (expandCmdDo t)
     withMacroExpansion (← getRef) cmd <| elabCommand cmd
   else if let some e := e? then
     let cmd := mkNullNode (expandCmdDo e)
     withMacroExpansion (← getRef) cmd <| elabCommand cmd
+
+@[implemented_by Meta.evalExpr]
+opaque evalExpr (α) (expectedType : Expr) (value : Expr) (safety := DefinitionSafety.safe) : MetaM α
+
+def toExprIO [ToExpr α] (x : IO α) : IO Expr :=
+  toExpr <$> x
+
+/--
+Executes a term of type `IO α` at elaboration-time
+and produces an expression corresponding to the result via `ToExpr α`.
+-/
+scoped syntax:lead (name := runIO) "run_io " term : term
+
+@[term_elab runIO]
+def elabRunIO : TermElab := fun stx expectedType? =>
+  match stx with
+  | `(run_io $t) => do
+    let expectedType := mkApp (mkConst ``IO) <|
+      expectedType?.getD <| ← mkFreshExprMVar <| mkSort <| .succ .zero
+    let v ← elabTermEnsuringType t expectedType
+    synthesizeSyntheticMVarsNoPostponing
+    let v ← instantiateMVars v
+    if (← logUnassignedUsingErrorInfos (← getMVars v)) then throwAbortTerm
+    let v ← mkAppM ``toExprIO #[v]
+    liftM <| ← evalExpr (IO Expr) (mkApp (mkConst ``IO) (mkConst ``Expr)) v
+  | _ => Elab.throwUnsupportedSyntax
+
+/-! ## ToExpr Instances -/
+
+instance : ToExpr System.FilePath where
+  toExpr p := mkApp (mkConst ``System.FilePath.mk) (toExpr p.toString)
+  toTypeExpr := mkConst ``System.FilePath
```

Comment:
Given that both `Lake` and `Std` need this now, perhaps it would be better to place in lean itself, to avoid duplication?

Reply:
I agree. I have just become somewhat wary of updating core unless necessary because it generally Sebastian or Leo to review and the are rather busy at the moment. This is probably small enough it does not matter. Regardless, it should probably be a separate PR because I did was move it around Lake in this one (it was previously in the deleted `Lake.Util.EvalTerm`).